src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp

```diff
@@ -0,0 +1,409 @@
+      {
+        String component_name = (String)features[feature_it].getSubordinates()[sub_it].getMetaValue("native_id"); 
+        std::cout << "component_name" << component_name << std::endl; //debugging
+        bool c_qc_pass = true;
+
+        // iterate through multi-feature/multi-sub-feature QCs/filters
+        // iterate through component_groups
+        for (size_t cg_qc_it = 0; cg_qc_it < filter_criteria.component_group_qcs_.size(); ++cg_qc_it)
+        {
+          if (filter_criteria.component_group_qcs_[cg_qc_it].component_group_name_ == component_group_name)
+          {
+            // labels and transition counts QC
+              std::cout << "n_heavy" << std::endl; //debugging
+            if (!checkRange(labels_and_transition_types["n_heavy"],
+              filter_criteria.component_group_qcs_[cg_qc_it].n_heavy_l_,
+              filter_criteria.component_group_qcs_[cg_qc_it].n_heavy_u_))
+            {
+              cg_qc_pass = false;
+            }
+            std::cout << "n_light" << std::endl; //debugging
+            if (! checkRange(labels_and_transition_types["n_light"],
+              filter_criteria.component_group_qcs_[cg_qc_it].n_light_l_,
+              filter_criteria.component_group_qcs_[cg_qc_it].n_light_u_))
+            {
+              cg_qc_pass = false;
+            }
+            std::cout << "n_detecting" << std::endl; //debugging
+            if (! checkRange(labels_and_transition_types["n_detecting"],
+              filter_criteria.component_group_qcs_[cg_qc_it].n_detecting_l_,
+              filter_criteria.component_group_qcs_[cg_qc_it].n_detecting_u_))
+            {
+              cg_qc_pass = false;
+            }
+            std::cout << "n_quantifying" << std::endl; //debugging
+            if (! checkRange(labels_and_transition_types["n_quantifying"],
+              filter_criteria.component_group_qcs_[cg_qc_it].n_quantifying_l_,
+              filter_criteria.component_group_qcs_[cg_qc_it].n_quantifying_u_))
+            {
+              cg_qc_pass = false;
+            }
+            std::cout << "n_identifying" << std::endl; //debugging
+            if (! checkRange(labels_and_transition_types["n_identifying"],
+              filter_criteria.component_group_qcs_[cg_qc_it].n_identifying_l_,
+              filter_criteria.component_group_qcs_[cg_qc_it].n_identifying_u_))
+            {
+              cg_qc_pass = false;
+            }
+            std::cout << "n_transitions" << std::endl; //debugging
+            if (! checkRange(labels_and_transition_types["n_transitions"],
+              filter_criteria.component_group_qcs_[cg_qc_it].n_transitions_l_,
+              filter_criteria.component_group_qcs_[cg_qc_it].n_transitions_u_))
+            {
+              cg_qc_pass = false;
+            }
+
+            // ion ratio QC
+            for (size_t sub_it2 = 0; sub_it2 < features[feature_it].getSubordinates().size(); ++sub_it2)
```

Comment:
I am kind of wondering how you encode the heavy/light in OpenMS, it seems that they are subfeatures of the same feature. Is that correct?

Reply:
Yes.  They are subfeatures of the same feature.  Basically I am using the "LabelType" tag in the TraML file to encode this.  In the case of metabolomics the "heavy" refers to the internal standard (i.e., there will most often be only 1 "heavy" transition per transition group).